### PR TITLE
Explicitly set ruby version in Gemfile

### PR DESCRIPTION
**Pull request by felixsch**
## Description

Adding an explicit version to the `Gemfile` should help `depfu` to not propose updates involving a major ruby change.

Fixes # (issue)

Depfu tries to upgrade to a newer major version of ruby

## Change Type

*Please select the correct option.*

- [x] **Bug Fix** (a non-breaking change which fixes an issue)
- [ ] **New Feature** (a non-breaking change which adds new functionality)
- [ ] **Documentation Update** (a change which only updates documentation)

## Checklist

*Please check off each item if the requirement is met.*

- [x] I have verified that my code follows RMT's coding standards with `rubocop`.
- [x] I have reviewed my own code and believe that it's ready for an external review.
- [x] I have provided comments for any hard-to-understand code.
- [x] I have documented the `MANUAL.md` file with any changes to the user experience.
- [x] RMT's test coverage remains at 100%.
- [x] If my changes are non-trivial, I have added a changelog entry to notify users at `package/obs/rmt-server.changes`.

## Other Notes
I will add this to the other pull requests changes. So there is no changes file here!

### CODE REVIEW

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/201802/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-build-images. (Merged c4eb9732fbcc6bc7f2d3dbd8119c3acd8caf4ca9 into bc06fd9f011ddf1959b0e472c44c868c72ea22ee)

**Comment by digitaltom** [comment]
The integration tests fail because on SLE12 SP2, there's already Ruby 2.5.9, but on SLE 15 GA and SP1, there is only 2.5.8.

Would it work to use `ruby '~> 2.5.8'`?

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/202269/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-build-images. (Merged c4eb9732fbcc6bc7f2d3dbd8119c3acd8caf4ca9 into 9228f9da8667575db38b9cab33b6596cf035a340)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/202270/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-build-images. (Merged f28442cda3a83600cebe00eaabfdd04ec9c5d39a into 9228f9da8667575db38b9cab33b6596cf035a340)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/202284/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-build-images. (Merged f28442cda3a83600cebe00eaabfdd04ec9c5d39a into 3aaec0e8c7e729f4ae431bccb0af6de7f9235ef9)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/202733/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-run-tests. (Merged 6e6966704d9e935d57f7970c3bee7d461f21fe97 into 3aaec0e8c7e729f4ae431bccb0af6de7f9235ef9)

**Comment by ngetahun** [comment]
It's failing because we test against different versions of ruby.

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/202759/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-run-tests. (Merged 6c2eb37acf0dd00420d21dd9c60fbcbfa7087bcb into 3aaec0e8c7e729f4ae431bccb0af6de7f9235ef9)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/202764/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-run-tests. (Merged 6c2eb37acf0dd00420d21dd9c60fbcbfa7087bcb into 7d8d070c942fd90d6e39d55a0a4600d472ec9f59)

**Comment by felixsch** [comment]
I'm wondering what is the best way forward here. We have this situation:

On the one hand we have `depfu` supplying is with updates for more recent versions of ruby (`2.6`, `2.7` etc.) but on the other hand we are bound to the version shipped in our SLE releases (2.5.8, 2.5.9). Now we also test for newer versions which break with my change.

@ngetahun Do you have any idea to solve this? Otherwise `depfu` is rather useless to use since we never can do the gems updates it provides since it will be always updates to versions not supported by SLE.

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/202765/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-run-tests. (Merged 3e39a1ef2c126c0c49695a82a41a2e1dad7abb0c into 7d8d070c942fd90d6e39d55a0a4600d472ec9f59)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/202784/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-run-tests. (Merged 3e39a1ef2c126c0c49695a82a41a2e1dad7abb0c into e249c059d27e400450264e8ba5d981ce4132e4fc)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/202886/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-run-tests. (Merged 3e39a1ef2c126c0c49695a82a41a2e1dad7abb0c into 50e5c7c3d9b1f94b3b3ea4d9a35ceaceffa5cef5)

**Comment by digitaltom** [comment]
I think we don't need to run github actions tests on Ruby versions which we don't release RMT for. Currently we only support RMT on 2.5.8 and 2.5.9.

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/203046/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-run-tests. (Merged 3e39a1ef2c126c0c49695a82a41a2e1dad7abb0c into 5dde35e84f00d3d7307a896d195004f6fd559dd7)

**Comment by felixsch** [comment]
I prepared a pull request for setting the test to `2.5.8` and `2.5.9`. https://github.com/SUSE/rmt/pull/750

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/203223/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-run-tests. (Merged 8cbded4824f5bb143ba1e6d73955fafad5fd4907 into 5dde35e84f00d3d7307a896d195004f6fd559dd7)

**Comment by ngetahun** [comment]
> I'm wondering what is the best way forward here. We have this situation:
> 
> On the one hand we have `depfu` supplying is with updates for more recent versions of ruby (`2.6`, `2.7` etc.) but on the other hand we are bound to the version shipped in our SLE releases (2.5.8, 2.5.9). Now we also test for newer versions which break with my change.
> 
> @ngetahun Do you have any idea to solve this? Otherwise `depfu` is rather useless to use since we never can do the gems updates it provides since it will be always updates to versions not supported by SLE.

Depfu reads `.ruby-version`, we can test against versions with GH actions but set the `.ruby-version` to `2.5.8`. A lasting solution for this would be to use multiple gemfiles (Each gemfile will contain a specific ruby version) together with [appraisal gem](https://github.com/thoughtbot/appraisal) for testing and set `.ruby-version` for depfu.

File: Gemfile [hunk 1]
```diff
@@ -1,5 +1,9 @@
 source 'https://rubygems.org'
 
+# Adding an explicit ruby version here to make sure it is still working with
+# SLE 15 variants, since we rely on the system installation of ruby
+ruby '~> 2.5.8'
```
Comment (ngetahun) [inline Gemfile:5]: Use `.ruby-version` file

File: Gemfile.lock [hunk 2]
```diff
@@ -402,5 +402,8 @@ DEPENDENCIES
   versionist
   webmock
 
+RUBY VERSION
+   ruby 2.5.8p224
```
Comment (ngetahun) [inline Gemfile.lock:406]: Use a `.ruby-version` file

**Review by ngetahun — CHANGES_REQUESTED** [review]

**Comment by ngetahun** [comment]
> I prepared a pull request for setting the test to `2.5.8` and `2.5.9`. #750

We should hold off merging this PR as this is just a patch and not a fix.

**Comment by felixsch** [comment]
I close this since this is not needed anymore!

**Pull request closed without merging**